components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/TenantSyncListener.java

```diff
@@ -0,0 +1,431 @@
+        // Not implemented.
+    }
+
+    /**
+     * Check whether tenant sharing is enabled and whether the creation is only a root organization.
+     *
+     * @param tenantId id of the tenant
+     * @return boolean Tenant creation firing is enable or not.
+     */
+    private boolean canCreateTenant(int tenantId) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Checking if tenant can be created. Tenant ID: " + tenantId);
+        }
+
+        boolean canCreateTenant = false;
+
+        IdentityEventListenerConfig identityEventListenerConfig = IdentityUtil.readEventListenerProperty(
+                UserOperationEventListener.class.getName(), TenantSyncListener.class.getName());
+
+        if (StringUtils.isNotBlank(identityEventListenerConfig.getEnable())
+                && Boolean.parseBoolean(identityEventListenerConfig.getEnable())) {
+            try {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+
+                Tenant tenant = realmService.getTenantManager().getTenant(tenantId);
+                String organizationID = tenant.getAssociatedOrganizationUUID();
+
+                // check if the Organization Depth in the Hierarchy is 0. only then create the root org.
+                if (StringUtils.isEmpty(organizationID) || IdentityMgtServiceDataHolder.getInstance().getOrganizationManager()
+                        .getOrganizationDepthInHierarchy(organizationID) == 0) {
+
+                    canCreateTenant = true;
+
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Tenant is a root organization and can be created. Tenant ID: "
+                                + tenantId);
+                    }
+
+                } else {
+                    LOG.debug("Skipping creating the tenant since the triggered Event is not related "
+                            + "to a root org creation.");
+                }
+
+                // if there was an exception thrown here, tenant activation won't happen.
+            } catch (UserStoreException | OrganizationManagementServerException e) {
+                LOG.error("Error while creating tenant ", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        } else {
+            LOG.debug(
+                    "Tenant sharing is disabled. Skipping tenant creation for tenant ID : " + tenantId);
+        }
+        return canCreateTenant;
+    }
+
+    /**
+     * Method to build the payload and send it to the external server. Event is sent asynchronously
+     *
+     * @param tenantInfo TenantInfoBean containing tenant details
+     * @param type       Type of the event
+     * @param eventURI   URI of the event
```

Comment:
```suggestion
     * Method to build the payload and send it to the external server. Event is sent asynchronously.
     *
     * @param tenantInfo TenantInfoBean containing tenant details.
     * @param type       Type of the event.
     * @param eventURI   URI of the event.
```

Reply:
fixed in https://github.com/wso2/carbon-identity-framework/pull/7075/commits/d32d019d0cf128847ec73bc433a30bb8d0ec34ee